### PR TITLE
[DOC] Add example on creating data table

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Users don't know how to create a Table.

##### Description of changes
Add an example how to create a Table from scratch.

##### Includes
- [ ] Code changes
- [ ] Tests
- [X] Documentation

### CODE REVIEW

File: doc/data-mining-library/source/reference/data.io.rst [hunk 1]
```diff
@@ -9,7 +9,6 @@ Loading and saving data (``io``)
 * Comma-separated values (\*.csv) file,
 * Tab-separated values (\*.tab, \*.tsv) file,
 * Excel spreadsheet (\*.xls, \*.xlsx),
```
Comment (janezd) [inline doc/data-mining-library/source/reference/data.io.rst:11]: Could you remove .xls here? We dropped support a while ago.

File: doc/data-mining-library/source/reference/data.io.rst [hunk 2]
```diff
@@ -56,7 +53,7 @@ A three-line header consists of:
    * ``weight`` (or ``w``) — the feature marks the weight of examples (in
      algorithms that support weighted examples),
    * ``ignore`` (or ``i``) — feature will not be imported,
-   * ``<key>=<value>`` custom attributes.
+   * ``<key>=<value>`` custom attributes. Such an example is ``type=image``, which denotes the path to the image (locally or online) or ``include=True``, which defines the variables as a text variable.
```
Comment (janezd) [inline doc/data-mining-library/source/reference/data.io.rst:56]: Maybe add some context here. `include=True` does not have any general meaning. E.g.


``<key>=<value>`` custom attributes recognized in specific contexts, for instance `color`, which defines the color palette when the variable is visualized, or `type=image` which signals that the variable contains a path to an image.

File: doc/data-mining-library/source/tutorial/data.rst [hunk 3]
```diff
@@ -11,7 +11,7 @@ Data Input
 ..  index::
     single: data; input
 
-Orange can read files in native tab-delimited format, or can load data from any of the major standard spreadsheet file types, like CSV and Excel. Native format starts with a header row with feature (column) names. The second header row gives the attribute type, which can be continuous, discrete, time, or string. The third header line contains meta information to identify dependent features (class), irrelevant features (ignore) or meta features (meta).
+Orange can read files in native tab-delimited format, or can load data from any of the major standard spreadsheet file types, like CSV and Excel. Native format starts with a header row with feature (column) names. The second header row gives the attribute type, which can be numeric, categorical, time, or string. The third header line contains meta information to identify dependent features (class), irrelevant features (ignore) or meta features (meta).
```
Comment (janezd) [inline doc/data-mining-library/source/tutorial/data.rst:14]: native -> proprietary

File: doc/data-mining-library/source/tutorial/data.rst [hunk 4]
```diff
@@ -60,6 +60,59 @@ The following script wraps-up everything we have done so far and lists first 5 d
 
 Note that data is an object that holds both the data and information on the domain. We show above how to access attribute and class names, but there is much more information there, including that on feature type, set of values for categorical features, and other.
 
+Creating a Data Table
+---------------------
+
+To create a data table from scratch, one needs two things, a domain and the data. The domain is the description of the variables, i.e. column names, types, roles, etc.
+
+First, we create the said domain. We will create three types of variable, numeric (ContiniousVariable), categorical (DiscreteVariable) and text (StringVariable). Numeric and categorical variables will be used a features (also known as X), while the text variable will be used as a meta variable.
```
Comment (janezd) [inline doc/data-mining-library/source/tutorial/data.rst:68]: variable -> variables?

File: doc/data-mining-library/source/tutorial/data.rst [hunk 5]
```diff
@@ -60,6 +60,59 @@ The following script wraps-up everything we have done so far and lists first 5 d
...
+
+    >>> from Orange.data import Domain, ContinuousVariable, DiscreteVariable, StringVariable
+    >>> 
+    >>> domain = Domain([ContinuousVariable("var1"), DiscreteVariable("var2", values=["red", "blue"])], metas=[StringVariable("var3")])
+
+Now, we will build the data with numpy.
+
+    >>> import numpy as np
+    >>> 
+    >>> var1 = np.array([1.2, 1.4, 1.5, 1.1, 1.2])
```
Comment (janezd) [inline doc/data-mining-library/source/tutorial/data.rst:78]: I'm thinking - would it be better to call this `column1` or `col1`? This is not a variable.

File: doc/data-mining-library/source/tutorial/data.rst [hunk 6]
```diff
@@ -60,6 +60,59 @@ The following script wraps-up everything we have done so far and lists first 5 d
 
 Note that data is an object that holds both the data and information on the domain. We show above how to access attribute and class names, but there is much more information there, including that on feature type, set of values for categorical features, and other.
 
+Creating a Data Table
+---------------------
+
+To create a data table from scratch, one needs two things, a domain and the data. The domain is the description of the variables, i.e. column names, types, roles, etc.
```
Comment (janezd) [inline doc/data-mining-library/source/tutorial/data.rst:66]: Add a reference to documentation of `Domain`.

File: doc/data-mining-library/source/tutorial/data.rst [hunk 7]
```diff
@@ -60,6 +60,59 @@ The following script wraps-up everything we have done so far and lists first 5 d
...
+
+Next, variables have to be transformed to columns (lists of lists). 
+
+    >>> var1 = var1.reshape(-1, 1)
+    >>> var2 = var2.reshape(-1, 1)
+    >>> var3 = var3.reshape(-1, 1)
+
+Finally, we create a table. We need a domain and variables, which can be passed as X (features), Y (class variable) or metas.
+
+    >>> table = Table.from_numpy(domain, X=X, metas=var3)
```
Comment (janezd) [inline doc/data-mining-library/source/tutorial/data.rst:94]: Where did X come from? `X = np.hstack((var1, var2, var3))`

However, there's no need to reshape. `X = np.column_stack((var1, var2, var3))` will stack 1-d arrays into columns.

**Review by janezd — COMMENTED** [review]
This documentation is indeed necessary. I have a couple of small comments.

File: doc/data-mining-library/source/reference/data.io.rst (hunk 1 above)
Comment (ajdapretnar) [reply doc/data-mining-library/source/reference/data.io.rst:11]: I tried it and it got loaded normally, this is why I kept it. 😮

File: doc/data-mining-library/source/reference/data.io.rst (hunk 2 above)
Comment (ajdapretnar) [reply doc/data-mining-library/source/reference/data.io.rst:56]: Perhaps we need to change the `include=True` into `text_feature=True` or something more descriptive. As far as I know, include is used exclusively for denoting text features.

Comment (ajdapretnar) [reply doc/data-mining-library/source/reference/data.io.rst:56]: @PrimozGodec What do you think?

File: doc/data-mining-library/source/tutorial/data.rst (hunk 5 above)
Comment (ajdapretnar) [reply doc/data-mining-library/source/tutorial/data.rst:78]: True, will fix it.

File: doc/data-mining-library/source/tutorial/data.rst (hunk 7 above)
Comment (ajdapretnar) [reply doc/data-mining-library/source/tutorial/data.rst:94]: Thanks, I was wondering which one is more appropriate to us. I have always been using hstack, but if column_stack does the trick, even better!

File: doc/data-mining-library/source/reference/data.io.rst (hunk 1 above)
Comment (ajdapretnar) [reply doc/data-mining-library/source/reference/data.io.rst:11]: io.py, L343. I am keeping it.

File: doc/data-mining-library/source/reference/data.io.rst (hunk 2 above)
Comment (ajdapretnar) [reply doc/data-mining-library/source/reference/data.io.rst:56]: I have kept @janezd 's proposal. Include should be one day renamed to a nicer pair, i.e. `type=text`.

**Comment by ajdapretnar** [comment]
@janezd This is ready for another review.

File: doc/data-mining-library/source/tutorial/data.rst [hunk 8]
```diff
@@ -60,6 +60,61 @@ The following script wraps-up everything we have done so far and lists first 5 d
...
+
+Now, we will build the data with numpy.
+
+    >>> import numpy as np
+    >>>
+    >>> column1 = np.array([1.2, 1.4, 1.5, 1.1, 1.2])
+    >>> column2 = np.array([0, 1, 1, 1, 0])
+    >>> column3 = np.array(["U13", "U14", "U15", "U16", "U17"], dtype=object)
+
+Two things to note here. var2 has values 0 and 1, even though we specified it will be a categorical variable with values "red" and "blue". X (features in the data) can only be numbers, so the numpy matrix will contain numbers, while Orange will handle the categorical representation internally. 0 will be mapped to the value "red" and 1 to "blue" (in the order, specified in the domain).
```
Comment (janezd) [inline doc/data-mining-library/source/tutorial/data.rst:85]: `var2` -> `column2`. Then I merge. :)

Comment (ajdapretnar) [reply doc/data-mining-library/source/tutorial/data.rst:85]: Sneaky devil, this "var". :)

**Pull request merged**